packages/bpk-component-banner-alert/src/AnimateAndFade.js

```diff
@@ -0,0 +1,162 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2017 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import BpkAnimateHeight from 'bpk-animate-height';
+import { durationBase } from 'bpk-tokens/tokens/base.es6';
+import { cssModules } from 'bpk-react-utils';
+import CSSTransitionGroup from 'react-transition-group/CSSTransitionGroup';
+
+import STYLES from './bpk-banner-alert.scss';
+
+const getClassName = cssModules(STYLES);
+
+const animateHeightDuration = parseInt(durationBase, 10);
+const fadeDuration = parseInt(durationBase, 10);
+
+class AnimateAndFade extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      atHeight: this.props.show,
+      visible: this.props.show,
+      hiding: false,
+      showing: false,
+    };
+
+    this.onToggle = this.onToggle.bind(this);
+    this.onFadeComplete = this.onFadeComplete.bind(this);
+    this.onAnimateHeightComplete = this.onAnimateHeightComplete.bind(this);
+    this.hide = this.hide.bind(this);
+    this.show = this.show.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.show !== this.props.show) {
+      this.onToggle();
+    }
+  }
+
+  onToggle() {
+    if (this.state.visible && this.state.atHeight) {
+      this.hide();
+    } else if (!this.state.visible && !this.state.atHeight) {
+      this.show();
+    }
+  }
+
+  onAnimateHeightComplete() {
+    if (this.state.showing) {
+      this.setState({
+        showing: false,
+      });
+    }
+  }
+
+  onFadeComplete() {
+    if (this.state.hiding) {
+      this.setState({
+        atHeight: false,
+        hiding: false,
+      });
+    }
+  }
+
+  show() {
+    this.setState({
+      atHeight: true,
+      showing: true,
+      visible: true,
+    });
+  }
+
+  hide() {
+    this.setState({
+      hiding: true,
+      visible: false,
+    });
+  }
+
+  render() {
+    const { children, onEnter, onLeave } = this.props;
+    const showPlaceholder = !this.state.visible && !this.state.hiding;
+
+    const fadingComponent = (
+      <CSSTransitionGroup
+        transitionName={{
+          leave: getClassName('bpk-banner-alert--leave'),
```

Comment:
These class names don't match the component, they should be something along the lines of `bpk-animate-and-fade--leave`